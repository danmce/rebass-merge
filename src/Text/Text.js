import React from 'react'
import PropTypes from 'prop-types'
import { Text as RebassText } from 'rebass'

const Text = ({ children, ...props }) => (
  <RebassText {...props}>{children}</RebassText>
)

Text.propTypes = {
  children: PropTypes.string,
  fontWeight: PropTypes.oneOf(['bold', 'normal']),
  color: PropTypes.string
}

export default Text
