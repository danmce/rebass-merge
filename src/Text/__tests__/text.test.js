import React from 'react'
import * as enzyme from 'enzyme'
import toJson from 'enzyme-to-json'
import Text from '../Text'

test('renders Text correctly', () => {
  const text = (
    <Text fontSize={[3, 4, 5]} fontWeight='bold' color='primary'>
      Text
    </Text>
  )

  expect(toJson(enzyme.mount(text))).toMatchSnapshot('enzyme.mount')
})
