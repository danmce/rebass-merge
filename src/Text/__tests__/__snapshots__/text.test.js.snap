// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Text correctly: enzyme.mount 1`] = `
<Text
  color="primary"
  fontSize={
    Array [
      3,
      4,
      5,
    ]
  }
  fontWeight="bold"
>
  <ForwardRef
    color="primary"
    fontSize={
      Array [
        3,
        4,
        5,
      ]
    }
    fontWeight="bold"
  >
    <Styled(div)
      color="primary"
      fontSize={
        Array [
          3,
          4,
          5,
        ]
      }
      fontWeight="bold"
      tx="text"
    >
      <div
        className="emotion-0"
      >
        Text
      </div>
    </Styled(div)>
  </ForwardRef>
</Text>
`;
